### ISSUE TITLE
Binding issues: Only binding to [::]:8554

### ISSUE THREAD

**Issue by JooJooBee666** [aler9/rtsp-simple-server#72]
No matter what I try here, I cannot access the streams from anything else but the local system.  I'm running this in a Raspberry Pi (Buster) 3b.  I can stream locally and read the stream locally, but that is it.  Netstat -ato shows me the following:

`tcp6       0      0 [::]:8554               [::]:*                  LISTEN      off (0.00/0/0)`

This is even after I disabled IPv6.  But,  this is the only thing actually listening on 8554.  I tried to use the --rtsp-port option but it looks like that argument, and all others, were removed at some other point.  How do I get this to bind to 0.0.0.0?

**Comment by aler9** [comment]
Hello,
1) please use a [template issue](https://github.com/aler9/rtsp-simple-server/issues/new/choose) when opening a issue, and provide the requested informations - next time i'm closing without replying
2) please provide the version you're using
3) command line options have been replaced with a configuration file, provided with the executable, since parameter nesting was needed
4) [::]:8554 means 0.0.0.0:8554, so it's all right. Try disabling the firewall, then force TCP by using the configuration file
```
protocols: [tcp]
```

Anyway, a log would be helpful.

**Comment by JooJooBee666** [comment]
Sorry about the template, didn't see it.  Anyhow, the version number is the latest 0.9.12.

However, it's working now.  I'm not sure what was going on earlier.  I ended up finding the option you mentioned for disabling IPv6 on the Pi before you replied and ran it but still had the same problems. Also, I was unable to figure out how to force any binding in the config, it's not clear in there either. 

Finally, at some point after countless reboots and other things, I just ran an apt update and apt upgrade again for the hell of it and rebooted (even though I just updated last week), then got busy with other stuff.  Came back to this and it was working, even after enabling IPv6 again.

Still strange, it looks like it's only binding to IPv6 even now, but it is definitely working over IPv4.  I guess netstat tupln isn't telling me the whole story.  Here's the truncated output now while it's working:

`tcp6       0      0 :::8554                 :::*                    LISTEN      13804/./rtsp-simple 
udp6       0      0 :::8000                 :::*                                13804/./rtsp-simple 
udp6       0      0 :::8001                 :::*                                13804/./rtsp-simple`

FYI, forcing TCP does not work with FFMPEG, it says "rtsp://192.168.1.249:8554/live: Protocol not found".  Here's the command I'm using, if you're curious: `raspivid -w 1648 -h 1232 -fps 30 -pf high -n -t 0 -o - | ffmpeg -hide_banner -r 30 -f h264 -i - -c:v copy -map 0:0 -r 30 -movflags frag_keyframe+empty_moov -f rtsp -b:v 2500k rtsp://192.168.1.249:8554/live`

Long story short, I can't reproduce the initial issue any longer so I guess it's safe to close.  Comments on the other items still appreciated. :)

**Comment by aler9** [comment]
I've added instructions on how to properly stream a Raspberry Pi Camere here: https://github.com/aler9/rtsp-simple-server#serve-a-raspberry-pi-camera
